samples/scala-eventsourced-counter/src/main/scala/com/example/domain/Counter.scala

```diff
@@ -25,6 +26,21 @@ class Counter(context: EventSourcedEntityContext) extends AbstractCounter {
         .thenReply(_ => Empty())
     }
 
+   import com.akkaserverless.javasdk.impl.DeferredCallImpl
+
+  override def increaseWithSideEffect(
+      currentState: CounterState,
+      increaseValue: example.IncreaseValue): EventSourcedEntity.Effect[Empty] =
+    if (increaseValue.value < 0) {
+      effects.error("Value must be a zero or a positive number")
+    } else {
+      effects
+        .emitEvent(ValueIncreased(increaseValue.value))
+        .thenAddSideEffect(_ =>
+          SideEffect(components.counter.increase(example.IncreaseValue(context.entityId, increaseValue.value * 2))))
+        .thenReply(_ => Empty())
+    }
+
```

Comment:
Would be great if you could add/modify a command handler that touches the context right away when handling the command as well for good measure, I think just logging the sequence number would be good enough.